src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -1807,6 +1877,160 @@ public override TypeKind TypeKind
             }
         }
 
+        private bool IsExtensionType([NotNullWhen(true)] out MethodDefinitionHandle? foundMarkerMethod)
+        {
+            var moduleSymbol = this.ContainingPEModule;
+            var module = moduleSymbol.Module;
+            foundMarkerMethod = null;
+
+            var isByRefLike = module.HasIsByRefLikeAttribute(_handle);
+            if (!isByRefLike)
+            {
+                return false;
+            }
+
+            foreach (var field in module.GetFieldsOfTypeOrThrow(_handle))
```

Comment:
>foreach (var field in module.GetFieldsOfTypeOrThrow(_handle)) [](http://example.com/codeflow?start=12&length=61)

Consider adding comments describing our intent. Checking for ..., because ...